src/main/java/core/basesyntax/strategy/FruitShopApplication.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.report.ReportGenerator;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.ShopService;
+import java.util.List;
+
+public class FruitShopApplication {
+
+    private final FileReader fileReader;
+    private final DataConverter dataConverter;
+    private final ShopService shopService;
+    private final ReportGenerator reportGenerator;
+    private final FileWriter fileWriter;
+
+    public FruitShopApplication(FileReader fileReader, DataConverter dataConverter,
+                                ShopService shopService, ReportGenerator reportGenerator,
+                                FileWriter fileWriter) {
+        this.fileReader = fileReader;
+        this.dataConverter = dataConverter;
+        this.shopService = shopService;
+        this.reportGenerator = reportGenerator;
+        this.fileWriter = fileWriter;
+    }
+
+    public void run(String inputFilePath, String outputFilePath) {

```

Comment:
This is a very general name, it is better to call it generateFruitReport or something similar